corehq/apps/domain/utils.py

```diff
@@ -111,3 +116,67 @@ def guess_domain_language(domain_name):
     domain = Domain.get_by_name(domain_name)
     counter = Counter([app.default_language for app in domain.applications() if not app.is_remote_app()])
     return counter.most_common(1)[0][0] if counter else 'en'
+
+
+@task(queue='background_queue')
+def send_repeater_payloads(repeater_id, payload_ids, email_id):
+    from corehq.motech.repeaters.models import Repeater, RepeatRecord
+    repeater = Repeater.get(repeater_id)
+    repeater_type = repeater.doc_type
+    payloads = dict()
+    headers = ['note']
+    result_file_name = "bulk-payloads-%s-%s-%s.csv" % (
+        repeater.doc_type, repeater.get_id,
+        datetime.datetime.utcnow().strftime("%Y-%m-%d--%H-%M-%S")
+    )
+
+    def get_payload(payload_id):
+        dummy_repeat_record = RepeatRecord(
+            domain=repeater.domain,
+            next_check=datetime.datetime.utcnow(),
+            repeater_id=repeater.get_id,
+            repeater_type=repeater_type,
+            payload_id=payload_id,
+        )
+        payload_doc = repeater.payload_doc(dummy_repeat_record)
+        payload = repeater.get_payload(dummy_repeat_record, payload_doc)
+        if isinstance(payload, dict):
+            return payload
+        else:
+            return json.loads(payload)
+
+    def populate_payloads():
+        for payload_id in payload_ids:
+            try:
+                payload = get_payload(payload_id)
+                payloads[payload_id] = payload
+                headers = list(set(headers + payload.keys()))
+            except Exception as e:
+                payloads[payload_id] = {'note': 'Could not generate payload, %s' % str(e)}
+
+    def create_result_file():
+        with open(result_file_name, 'w') as csvfile:
+            headers.append('payload_id')
+            writer = csv.DictWriter(csvfile, fieldnames=headers)
+            writer.writeheader()
+            for payload_id, payload in payloads.items():
+                row = payload
+                row['payload_id'] = payload_id
+                writer.writerow(row)
+
+    def email_payloads():
+        create_result_file()
+        with open(result_file_name) as csvfile:
+            email = EmailMessage(
+                subject="Bulk Payload Report [%s]" % repeater_type,
+                body="PFA report",
+                to=[email_id],
+                from_email=settings.DEFAULT_FROM_EMAIL
+            )
+            email.attach(filename=result_file_name, content=csvfile.read())
+            csvfile.close()
+            email.send()
+            os.remove(result_file_name)
```

Comment:
@esoergel 
thought about this a bit. You suggested using a Temp file..
If i just all content in memory and then just email it, the only concern was if the content is quite large, it would take quite some space in memory. So kept it in the file only and removed it post email.
Let me know if there are any concerns with this.

Reply:
I was actually just looking at this same sort of thing with Jordan.  The locations export is the simplest example I saw: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/locations/util.py#L283-L302
This uses a temporary file location to stream to while writing the file, then saves it in Riak for a timeout period and provides a location where the user can download it.  The nice thing about that approach is that you don't have to send any potentially sensitive info over email.

It depends on what this is going to be used for though.  This is probably fine as long as you know the contents of the file aren't going to contain sensitive info.  If they might, we should make an effort to keep that to secured channels.